### PR TITLE
Ensure that Prolific API token is sent to the deployed server

### PR DESCRIPTION
**Pull request by pmcharrison**
Fixes a bug where the Prolific API token wouldn't be sent to the deployed server unless it was specified in config.txt rather than ~/.dallingerconfig. Also fixed a bug where SMTP authentication details weren't being sent to Docker-SSH deployments.

### CODE REVIEW

**Review by fmhoeger — APPROVED** [review]

**Comment by pmcharrison** [comment]
Sorry, the diff looks weird as it includes some commits that already went to master! Not sure why that is....

File: .bumpversion.cfg [hunk 1]
```diff
@@ -1,5 +1,5 @@
 [bumpversion]
-current_version = 9.4.2
+current_version = 9.5.0a1
```
Comment (pmcharrison) [inline .bumpversion.cfg:2]: You can ignore this, it's already on master

File: dallinger/prolific.py [hunk 2]
```diff
@@ -239,7 +244,13 @@ def _req(self, method: str, endpoint: str, **kw) -> dict:
...
             return {"status_code": response.status_code}
 
-        parsed = response.json()
+        try:
+            parsed = response.json()
+        except requests.exceptions.JSONDecodeError as err:
+            raise ProlificServiceException(
+                f"Failed to parse the following JSON response from Prolific: {err.doc}"
+            )
+
```
Comment (pmcharrison) [inline dallinger/prolific.py:253]: You can ignore this, it's already on master

File: dallinger/recruiters.py [hunk 3]
```diff
@@ -292,6 +292,10 @@ def normalize_entry_information(self, entry_information: dict):
 
     def recruit(self, n: int = 1):
         """Recruit `n` new participants to an existing Prolific Study"""
+        if not self.config.get("auto_recruit"):
+            logger.info("auto_recruit is False: recruitment suppressed")
+            return
+
```
Comment (pmcharrison) [inline dallinger/recruiters.py:298]: You can ignore this, it's already on master

File: dallinger/version.py [hunk 4]
```diff
@@ -1,3 +1,3 @@
 """Dallinger version number."""
 
-__version__ = "9.4.2"
+__version__ = "9.5.0a1"
```
Comment (pmcharrison) [inline dallinger/version.py:3]: You can ignore these version number changes, they are already on master

**Comment by fmhoeger** [comment]
@pmcharrison The first thing we always need to do after releasing a  new version is merging the "Bump version on master branch post-release" into master before continuing with development. This time there was some delay. I think this is why it happened.

**Pull request merged**